felix/dataplane/linux/endpoint_mgr.go

```diff
@@ -658,13 +659,9 @@ func (m *endpointManager) resolveWorkloadEndpoints() {
 					natInfos = workload.Ipv6Nat
 					addrSuffix = "/128"
 				}
-				if m.floatingIPsEnabled && len(natInfos) != 0 {
-					// Include any floating IP NATs if the feature is enabled.
-					old := ipStrings
-					ipStrings = make([]string, len(old)+len(natInfos))
-					copy(ipStrings, old)
-					for ii, natInfo := range natInfos {
-						ipStrings[len(old)+ii] = natInfo.ExtIp + addrSuffix
+				for _, natInfo := range natInfos {
+					if m.floatingIPsEnabled || natInfo.Source == apiv3.IPNATSourceOpenStack {
+						ipStrings = append(ipStrings, natInfo.ExtIp+addrSuffix)
```

Comment:
Not sure if this is OK, appending to a slice can share the same backing array so if two bits of code decide to append different things to the same slice then they might see each other's updates.  Better to take your own copy?

Reply:
Yes, I've restored logic like we had here before.